lms/djangoapps/courseware/views.py

```diff
@@ -1577,12 +1594,11 @@ def financial_assistance_form(request):
     enrolled_courses = [
         {'name': enrollment.course_overview.display_name, 'value': unicode(enrollment.course_id)}
         for enrollment in CourseEnrollment.enrollments_for_user(user).order_by('-created')
-        if CourseMode.objects.filter(
+        if enrollment.mode != CourseMode.VERIFIED and CourseMode.objects.filter(
```

Comment:
It would be more readable, but this way we could avoid a database query if the first condition is not satisfied.